src/main/java/blackjack/model/card/Card.java

```diff
@@ -0,0 +1,40 @@
+package blackjack.model.card;
+
+import java.util.Objects;
+
+public final class Card {
+
+    private final Suit suit;
+    private final CardValue cardValue;
+
+    Card(Suit suit, CardValue cardValue) {
+        this.suit = suit;
+        this.cardValue = cardValue;
+    }
+
+    public String getDisplayLabel() {
+        return cardValue.getLabel() + suit.getLabel();
+    }
+
+    public CardValue getCardValue() {
+        return cardValue;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Card card = (Card) o;
+        return suit == card.suit
+                && cardValue == card.cardValue;
+    }
```

Comment:
```suggestion
    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Card card)) return false;
        return suit == card.suit && cardValue == card.cardValue;
    }
```
저는 로직이 간결해서 instanceof 기반의 equals 로직을 애용하고 있어요~ 벨로는 어떻게 생각하시나요?
<img width="464" alt="image" src="https://github.com/user-attachments/assets/d9a1a2a6-a357-4744-b7bf-6391954538dd" />

Reply:
상속을 염두하지 않았다면 `getClass()` 방식을 사용하는 것이 더 적절하다고 생각했어요.
그 이유는 `같은 타입의 객체끼리만 비교해야 한다` 는 의도를 조금 더 선명하게 전달할 수 있지 않을까? 였어요.

상속을 고려하지 않는 클래스라면 `final` 키워드를 사용해서 하위 타입이 생성되지 않도록 제한하는 것과 비슷하게 `final`과 함께 `getClass()`를 사용하면, `이 클래스는 확장되지 않으며, 같은 타입의 인스턴스만 비교할 수 있다`는 점을 코드로 표현하는 것 같았어요.

개인적으로도 상속을 적극적으로 활용한 경험이 많지 않다 보니,
실제 코드를 작성할 때도 `같은 타입끼리만 equals() 비교`를 수행하는 경우가 대부분이었어요.
그러다 보니 다형성을 고려해야 하는 상황 (instanceof 방식이 필요한 경우)이 거의 없었고,
자연스럽게 조금 더 명시적인 `getClass()` 방식을 선호하게 된 것 같아요.

비교해보니 `instanceof 방식`의 로직이 더 간결하네요! 👍 👍 
지금까지 `equals()`를 작성할 때 `간결함`을 기준으로 선택한 적은 없었는데,
아무래도 `equals()`는 보통 하단에 선언해 두고 자주 들여다볼 일이 없어서 그런 것 같아요.

혹시 이프는 `equals()`를 자주 변경해야 했던 경험이 있거나,
하위 타입을 고려해야 했던 사례가 있나요? 궁금해요! 😊